### DISCUSSION TITLE
macOS + KiCad 8 libraries not working

### DISCUSSION THREAD

**Discussion by blark** [xesscorp/skidl#223] [Q&A] [answered]
I have installed skidl in a venv (3.12.3), and can't get search working. Any advice would be appreciated.

```
❯ python3
Python 3.12.3 (main, Apr  9 2024, 08:09:14) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from skidl import *
WARNING: KICAD6_SYMBOL_DIR environment variable is missing, so the default KiCad symbol libraries won't be searched. @ [/Users/blark/src/lab/skidl/<frozen importlib._bootstrap_external>:995=>/Users/blark/src/lab/skidl/<frozen importlib._bootstrap>:488]
WARNING: KICAD7_SYMBOL_DIR environment variable is missing, so the default KiCad symbol libraries won't be searched. @ [/Users/blark/src/lab/skidl/<frozen importlib._bootstrap_external>:995=>/Users/blark/src/lab/skidl/<frozen importlib._bootstrap>:488]
WARNING: KICAD_SYMBOL_DIR environment variable is missing, so the default KiCad symbol libraries won't be searched. @ [/Users/blark/src/lab/skidl/<frozen importlib._bootstrap_external>:995=>/Users/blark/src/lab/skidl/<frozen importlib._bootstrap>:488]
>>> print(lib_search_paths)
{'kicad8': ['.', '/Applications/KiCad/KiCad.app/Contents/SharedSupport/symbols/'], 'kicad6': ['.'], 'kicad7': ['.'], 'spice': ['.'], 'kicad5': ['.'], 'skidl': ['.', '/Users/blark/src/lab/skidl/venv/lib/python3.12/site-packages/skidl/tools/skidl/libs']}
>>> search('opamp')
>>>
```

**Comment by devbisme** [comment, marked as answer]
The problem is `skidl` defaults to using KiCad 5 as the default backend tool. KiCad 5 uses part libraries that end with `.lib`. Since you're using KiCad 8 (I assume), your libraries end with `.kicad_sym` and `search` won't see them so it finds nothing.

The solution is to use either `search("opamp", tool=KICAD8)` or set the default tool to KiCad 8 using `set_default_tool(KICAD8)` and then run `search("opamp")`.

**Reply by blark** [reply]
Thanks for the quick response! I’ll try that, appreciate you helping out here as I probably should have read the code to see what was going on… for what it’s worth I did refer to the “basic usage” docs, it might be worth adding this very helpful tidbit there (I see that the docs are in the repo, I’ll submit a PR).
